### PR TITLE
ROX-19773: API token expiration events

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

This PR enables the creation of administration events for the log messages for API token expirations.

This enables more visiblity for the expiration.

A new resource `API Token` was created with the new domain `Authentication`.

Additionally provided a fix to allow passing both `resource ID` and `resource name` fields via zap.Field. Previously this rendered in an error.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- see the added unit tests.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7869** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7869" target="_blank">Graphite</a> 
    * **PR #7941** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7941" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/7941?utm_source=stack-comment).

**Review by stehessel — APPROVED** [review]

File: pkg/administration/events/hints.go [hunk 1]
```diff
@@ -15,19 +15,30 @@ var (
 	// In the future, we may extend this, and possibly also ensure hints are loaded externally (similar to
 	// vulnerability definitions).
 	hints = map[string]map[string]map[string]string{
+		authenticationDomain: {
+			adminResources.APIToken: {
+				"": `An API token is about to expire. See the details on the expiration time within the event message.
+It is not possible to re-create the token, instead you have to do the following:
```
Comment (kcarmichael08) [inline pkg/administration/events/hints.go:21]: Suggestion:

You cannot re-create the token. Instead, perform these steps:

File: pkg/administration/events/hints.go [hunk 2]
```diff
@@ -15,19 +15,30 @@ var (
...
+- Delete the expiring API token.
+- Create a new API token (you may choose the same name).
+
+Afterwards, you may use the newly created API token.
```
Comment (kcarmichael08) [inline pkg/administration/events/hints.go:25]: suggestion:

You can then use the newly-created API token.

(we don't use afterwards or may)

File: pkg/administration/events/hints.go [hunk 3]
```diff
@@ -15,19 +15,30 @@ var (
 	// In the future, we may extend this, and possibly also ensure hints are loaded externally (similar to
 	// vulnerability definitions).
 	hints = map[string]map[string]map[string]string{
+		authenticationDomain: {
+			adminResources.APIToken: {
+				"": `An API token is about to expire. See the details on the expiration time within the event message.
+It is not possible to re-create the token, instead you have to do the following:
+- Delete the expiring API token.
+- Create a new API token (you may choose the same name).
```
Comment (kcarmichael08) [inline pkg/administration/events/hints.go:23]: Create a new API token (you can choose the same name).

(we don't use "may")

**Review by kcarmichael08 — COMMENTED** [review]
These are great! I noted a couple of things inline and have a couple of general comments:

- We [don't use "please"](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures) in technical documentation.
- [Avoid using slashes](https://www.ibm.com/docs/en/ibm-style?topic=punctuation-slashes) (example: registry/repository) - use "and" or "or" (example: registry or repository).

**Pull request merged**